qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/vqe_ucc_factory.py

```diff
@@ -33,23 +38,60 @@
 from ...initial_points import InitialPoint, HFInitialPoint
 from .minimum_eigensolver_factory import MinimumEigensolverFactory
 
+logger = logging.getLogger(__name__)
 
-class VQEUCCFactory(MinimumEigensolverFactory):
-    """A factory to construct a VQE minimum eigensolver with UCCSD ansatz wavefunction."""
 
+class VQEUCCFactory(MinimumEigensolverFactory):
+    """Factory to construct a :class:`VQE` minimum eigensolver with :class:`UCCSD` ansatz wavefunction.
+
+    Note: Changing the ansatz of the minimum_eigensolver of the factory won't affect
+        the ansatz of new solvers produced by the factory. On top of that the default
+        type for :class:`VQE` is not the same as the factory.
+
+    .. code-block:: python
+
+        #Wrong way
+        factory = VQEUCCFactory()
+        print(type(factory.minimum_eigensolver.ansatz)) # RealAmplitudes (by default)
+        factory.minimum_eigensolver.ansatz = UCC()
+        print(type(factory.minimum_eigensolver.ansatz)) # UCC
+        vqe = factory.get_solver(problem,qubit_converter)
+        print(type(vqe.ansatz))  # UCCSD (and not UCC nor RealAmplitudes)
+        #Expected way
+        factory = VQEUCCFactory()
+        vqe1 = factory.get_solver(problem,qubit_converter)
+        print(type(vqe1.ansatz))    #UCCSD (default)
+        factory.ansatz = UCC()
+        vqe2 = factory.get_solver(problem,qubit_converter)
+        print(type(vqe2.ansatz))    #UCC
```

Comment:
Just to be consistent with our black code formatter even here:
```suggestion
        # Wrong way
        factory = VQEUCCFactory()
        print(type(factory.minimum_eigensolver.ansatz))  # RealAmplitudes (by default)
        factory.minimum_eigensolver.ansatz = UCC()
        print(type(factory.minimum_eigensolver.ansatz))  # UCC
        vqe = factory.get_solver(problem, qubit_converter)
        print(type(vqe.ansatz))   # UCCSD (and not UCC nor RealAmplitudes)
        # Expected way
        factory = VQEUCCFactory()
        vqe1 = factory.get_solver(problem, qubit_converter)
        print(type(vqe1.ansatz))  # UCCSD (default)
        factory.ansatz = UCC()
        vqe2 = factory.get_solver(problem, qubit_converter)
        print(type(vqe2.ansatz))  # UCC
```